vms/avm/txs/mempool/mempool.go

```diff
@@ -187,6 +187,18 @@ func (m *mempool) Peek() (*txs.Tx, bool) {
 	return tx, exists
 }
 
+func (m *mempool) Iterate(f func(*txs.Tx) bool) {
+	m.lock.RLock()
+	defer m.lock.RUnlock()
```

Comment:
The lock is needed here to honor the `Iter` invariant that the linked hash map isn't arbitrarily modified while iterating.